scripts/doc-coverage.py

```
@@ -0,0 +1,405 @@
+    except json.JSONDecodeError as error:
+        detail = (
+            f"cargo rustdoc for {target.package} {target.kind}"
+            f" ({target.name or 'lib'}) did not emit coverage JSON: {error}"
+        )
+        raise RuntimeError(detail) from error
+    if not isinstance(per_file, dict):
+        detail = (
+            f"cargo rustdoc for {target.package} {target.kind}"
+            f" ({target.name or 'lib'}) did not emit coverage JSON: expected an object"
+        )
+        raise RuntimeError(detail)
```

Ruff (0.16.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)